python/src/numerous/_client/_fs_client.py

```diff
@@ -101,6 +173,7 @@ class FileSystemClient:
     def __init__(self, base_path: Path) -> None:
         self._base_path = base_path
         self._base_path.mkdir(exist_ok=True)
+        self.file_to_path_map: dict[str, Path] = {}
```

Comment:
For the `FileSystemClient` we want to store all state in the file system. Currently files are stored in two places - in the root (_base_path / file_id), and under the collection.

I think we should store a file index on the file system, e.g. a file `base_path/__file_index/<file_id>` would contain the path relative to base path `<collection_id>/file_<file_id>`.

Reply:
At least in the current implementation, it would not recognize files from previous sessions, since they are not added to the `file_to_path_map`.

Basically we are just storing the same information on the file system.